woocommerce/woocommerce
# Add shipping class data store

### All Submissions:






### Changes proposed in this Pull Request:

Adds a new CRUD data store for the product shipping classes API rest endpoint.

Closes #33456 & #33457 & #33458

### How to test the changes in this Pull Request:

1. Load this branch and go to **WooCommerce > Home** and open your console
2. Run `const actions = wp.data.dispatch('experimental/wc/admin/products/shipping-classes');`
3. Create a new shipping class by running: `actions.createProductShippingClass( { name: 'Test', slug: 'test' } );`
4. See if the new shipping class was created by using the selectors:
```
const selectors = wp.data.select('experimental/wc/admin/products/shipping-classes');
selectors.getProductShippingClasses({}); // run this a second time after the request resolved to see the output.
```
5. Now get the `id` of newly created shipping class and update it: `actions.updateProductShippingClass( 12,  { name: 'Test updated' } );`
6. Check the network panel and run `selectors.getProductShippingClasses({});` again to see if it updated correctly.
7. Refresh the page set the `actions` & `selectors` const again as shown above. Now only get the shipping class by the `id`: `selectors.getProductShippingClass(12);`, run a second time to see the output in the console.
8. Now delete the product shipping class with `actions.deleteProductShippingClass( 12 );` and run `selectors.getProductShippingClasses({});` to see if it doesn't exist anymore.
9. Now try using some of the query params for `selectors.getProductShippingClasses({});` that are listed [here](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-shipping-classes) (this might require you to create some new shipping classes again.

### Other information:




### FOR PR REVIEWER ONLY:
